pkg/validations/upgradevalidations/immutableFields.go

```diff
@@ -20,6 +20,12 @@ func ValidateImmutableFields(ctx context.Context, k ValidationsKubectlClient, cl
 		return fmt.Errorf("cluster name is immutable")
 	}
 
+	if prevSpec.Namespace != spec.Namespace {
+		if !(prevSpec.Namespace == "default" && spec.Namespace == "") {
```

Comment:
Do we need to have this same check for GitOps and OIDC? Or is this not the case because it would fail in validations earlier anyways? If that's the case, we technically don't need the checks above for VSphere if this check here covers it for all the config anyways.

Reply:
That's true. These cases are covered in earlier validations. I removed the unnecessary checks